### PR TITLE
Fixes and updates the TEG

### LINKED ISSUE #75738: The thermo-electric generator (TEG) does not work.
Reporting client version: 514.1589

## Round ID:
[207199](https://scrubby.melonmesa.com/round/207199)

## Testmerges:
- [Adds a new front end for viewing logs | fixes the manifest log message too](https://www.github.com/tgstation/tgstation/pull/75617)

## Reproduction:
Spawn a TEG and try to set it up. Far as I can tell gas does not flow through it.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Fixes thermoelectric generators so they work again
Pipes can be connected to it, the machine can be rotated
Adds the ventcrawl movement flags as a defined bitfield
I also gave it a TGUI menu, it was small so it is pretty insignificant. I added a little bit more text to error messages to make it clearer why the thermoelectric generator isn't functional.
I also repathed generator to thermoelectric_generator because 'generator' is a keyword and is highlighted in green which makes people using vscode a little confused what it's meant to be.

Old
image

New
https://github.com/tgstation/tgstation/assets/53777086/064a5dda-5407-4817-b090-d22eb6c4aab8

## Why It's Good For The Game

This is one of the things I had to move to TGUI in https://hackmd.io/XLt5MoRvRxuhFbwtk4VAUA
I was originally gonna remove it, but in the spirit of feature freeze I thought I should at least give it a try.
This fixes many issues with it and gives it a new better UI that won't stop updating easily so you can actually watch the changes as it happens.
The TEG may not be obtainable in-game but it can still be mapped in or give by admins, letting it function as intended is still a massive benefit.

Closes https://github.com/tgstation/tgstation/issues/75738

## Changelog

:cl: JohnFulpWillard, Unit2E teaching me the TEG
fix: The TEG now works again (still unobtainable by regular means though).
fix: the TEG and its circulators can now be rotated counterclockwise again.
refactor: The TEG now uses a TGUI interface rather than the old HTML one.
/:cl:

### CODE REVIEW

**Comment by rageguy505** [comment]
Probably fixed #75738

**Comment by 13spacemen** [comment]
@JohnFulpWillard 
can you please also port these amazing TEG sprites
https://github.com/yogstation13/Yogstation/pull/9195

**Comment by LemonInTheDark** [comment]
if you update the teg sprites retain the old one for use in the nostalgia shuttle

**Comment by JohnFulpWillard** [comment]
when i said to add it i meant ur own pr

File: code/modules/power/thermoelectric_generator.dm [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	find_circulators()
+	balloon_alert(user, "circulators updated")
+	return TRUE
+
+/obj/machinery/power/thermoelectric_generator/screwdriver_act(mob/user, obj/item/I)
+	if(!anchored)
+		balloon_alert(user, "anchor it down!")
+		return
+	toggle_panel_open()
+	I.play_tool_sound(src)
```
Comment (Ghommie) [inline code/modules/power/thermoelectric_generator.dm:85]:
```suggestion
/obj/machinery/power/thermoelectric_generator/screwdriver_act(mob/user, obj/item/tool)
	if(!anchored)
		balloon_alert(user, "anchor it down!")
		return
	toggle_panel_open()
	tool.play_tool_sound(src)
```

File: code/modules/power/thermoelectric_generator.dm [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	null_circulators()
+
+/obj/machinery/power/thermoelectric_generator/update_overlays()
+	. = ..()
+	if(machine_stat & (NOPOWER|BROKEN))
+		return
+
+	var/L = min(round(lastgenlev / 100000), 11)
+	if(L != 0)
+		. += mutable_appearance('icons/obj/machines/engine/other.dmi', "[base_icon_state]-op[L]")
```
Comment (Ghommie) [inline code/modules/power/thermoelectric_generator.dm:55]:
```suggestion
	var/level = min(round(lastgenlev / 100000), 11)
	if(level)
		. += mutable_appearance('icons/obj/machines/engine/other.dmi', "[base_icon_state]-op[level]")
```

**Comment by LemonInTheDark** [comment]
I am old

**Pull request merged**